app/helpers/gamification/application_helper.rb

```diff
@@ -16,6 +16,9 @@ def reward rewardable, options
         concat hidden_field_tag 'reward[rewarding_id]', rewarding.id
         concat hidden_field_tag 'reward[rewardable_type]', rewardable.class.name
         concat hidden_field_tag 'reward[rewardable_id]', rewardable.id
+        concat hidden_field_tag 'checksum', Checksum.generate(
```

Comment:
That's not stupid. I just put it there because the `reward` bit is a convention that it refers to the `reward` model, so everything inside that are attributes that you can expect to find on that model. `checksum` isn't an attribute on that model, though.